File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let answer = s;
    let zeroSum = 0;
    let count = 0;
    
    while(answer.length > 1){
        let zero = 0;
        let noZero = 0;

        for(let value of answer){
            if(value === "0") zero++;
            else noZero++;
        }
        
        answer = noZero.toString(2);
        zeroSum += zero;
        count++;
    }
    
    return [count, zeroSum];
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int count = 0;
        int removedZero = 0;
        
        while(s.length() > 1){
            int beforeRemoveZero = s.length();
            s = s.replaceAll("0","");
            int afterRemoveZero = s.length();
            s = Integer.toBinaryString(s.length());
            count++;
            removedZero += beforeRemoveZero - afterRemoveZero;
        }
        
        answer[0] = count;
        answer[1] = removedZero;
        
        return answer;
    }
}